src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java

```diff
@@ -142,6 +162,50 @@ private String generateQuery(String partitionFromDate, String partitionToDate, S
     return String.format(queryTemplate, tableName, condition.toString());
   }
 
+  private String generateQueryForMaterializingView(String dataset, String table, String filter) {
+    String queryTemplate = "select * from %s %s";
+    StringBuilder condition = new StringBuilder();
+
+    if (!Strings.isNullOrEmpty(filter)) {
+      condition.append(String.format(" where %s", filter));
+    }
+
+    String tableName = dataset + "." + table;
```

Comment:
Should this also include project id?

Reply:
@nitinmotgi we did not add the `datasetProjectId` here because it is already set in method `runQuery` on line 119